File: typescript/src/n0704_binary_search.test.ts
import { expect } from "jsr:@std/expect";
import { binarySearch } from "./n0704_binary_search.ts";

Deno.test("binarySearch", () => {
  expect(binarySearch([-1, 0, 3, 5, 9, 12], 9)).toEqual(4);
  expect(binarySearch([-1, 0, 3, 5, 9, 12], 2)).toEqual(-1);
});

File: rust/src/n0704_binary_search.test.rs
use crate::n0704_binary_search::*;

#[test]
fn test_binary_search() {
    assert_eq!(binary_search(&[-1, 0, 3, 5, 9, 12], 9), 4);
    assert_eq!(binary_search(&[-1, 0, 3, 5, 9, 12], 2), -1);
}
